### PR TITLE
Sample for opo-rub

### PR DESCRIPTION
**Pull request by ohler55**
The sample was updated so it can be run with opo-rub and embedded Ruby.
The wab.pathPrefix was added to the Javascript to allow both IO and
embedded Ruby controllers to be run at the same time.

### CODE REVIEW

File: examples/sample/README.md [hunk 1]
```diff
@@ -59,6 +60,20 @@ Near the bottom of the conf file, the Controller `spawn.rb` is mentioned
 along with command line options. The `-v` option turns on Controller
 verbosity.
 
+### OpO-Rub
+
+OpO-Rub provides the HTTP server and the Model storage, and is able to run the
+an an embedded Ruby application.
```
Comment (ashmaroli) [inline examples/sample/README.md]: typo in this line..

File: examples/sample/README.md [hunk 2]
```diff
@@ -59,6 +60,20 @@ Near the bottom of the conf file, the Controller `spawn.rb` is mentioned
...
+
+OpO-Rub can be downloaded from
+[OpO Downloads](http://www.opo.technology/download/index.html).
+
+The confguration file for OpO-Rub is in the `opo` sub-directory, and is named
```
Comment (ashmaroli) [inline examples/sample/README.md]: typo: `confguration`

File: examples/sample/opo/embed.conf [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+#dir = /var/opo
+dir = examples/sample/opo/data
+
+# A journal is kept if this option is true. When set to false changes in the
+# data are not persisted until a clean exit or a explicit persist command to
+# the opod.
+journal = false
+
+# If sync is true then writes are always synced to disk before
+# returning. Performance is far better is sync is set to false.
```
Comment (ashmaroli) [inline examples/sample/opo/embed.conf]: typo: `is far better is sync`

File: examples/sample/opo/embed.conf [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+sync = false
+
+# If exit_sync is true then writes are performed on normall shutdown or by
```
Comment (ashmaroli) [inline examples/sample/opo/embed.conf]: typo: `normall`

File: examples/sample/opo/embed.conf [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+# Size that when exceeded cause the log files to rotate. Note file may and are
+# usually slightly larger than the maximum size as the the maximum size is a
+# trigger to rotate.
+log.maxSize = 100000000
+
+# If true log entries are displayed on the console.
+log.console = true
+
+# if true console log out is in classic format. If false output is in JSON
```
Comment (ashmaroli) [inline examples/sample/opo/embed.conf]:
```diff
- console log out
+ console log output
```

File: examples/sample/opo/embed.conf [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+# format.
+log.classic = true
+
+# If true console output is colorized.
+log.colorize = true
+
+# If the syslog address and port in the format 10.10.0.7:1234 is provided log
+# messages are also send to syslog on the provided address. If the UDP option
```
Comment (ashmaroli) [inline examples/sample/opo/embed.conf]:
```diff
- are also send to syslog
+ are also sent to syslog
```

File: examples/sample/opo/embed.conf [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+handler.embed.path = /v1/**
+# An embedded handler is denoted by a class value.
+handler.embed.class = SampleController
+
+################################################################################
+# When using opo-rub or other opo version with embedded interpreters the
+# 'embed' options can be used. The handler options are also extended to
+# support calling the embedded interpreter.
+
+embed.load_path = .,lib,examples/sample
```
Comment (ashmaroli) [inline examples/sample/opo/embed.conf]: prefer a space after comma for readability:
```diff
- embed.load_path = .,lib,examples/sample
+ embed.load_path = ., lib, examples/sample
```
or preferably denoted as an Array with `[]`

File: script/start-sample [hunk 8]
```diff
@@ -34,7 +34,7 @@ set -e
 
 sampledir=examples/sample
 
-while getopts :bo opt
+while getopts :boe opt
```
Comment (ashmaroli) [inline script/start-sample]: prefer adding a comment explaining what `:boe` is..

Comment (ohler55) [reply script/start-sample]: uhm, this is your script. I can add a reference to the bash man pages. Maybe you meant to add a comment in the usage as to what the -e option is.

File: examples/sample/opo/embed.conf (hunk 7 above)
Comment (ohler55) [reply examples/sample/opo/embed.conf]: I'll make a note of that for OpO. Can't do anything about it here.

File: script/start-sample (hunk 8 above)
Comment (ashmaroli) [reply script/start-sample]: Ah.. right.. yes..

File: examples/sample/README.md [hunk 9]
```diff
@@ -63,12 +63,12 @@ verbosity.
 ### OpO-Rub
 
 OpO-Rub provides the HTTP server and the Model storage, and is able to run the
```
Comment (ashmaroli) [inline examples/sample/README.md]: the `the` is superfluous

File: script/start-sample (hunk 8 above)
Comment (ashmaroli) [reply script/start-sample]: (the reference to bash man pages is not required..)

File: examples/sample/README.md [hunk 10]
```diff
@@ -173,35 +187,74 @@ connections at a time to the Runner.
...
-with an average latency of 4.249 msecs
+127.0.0.1:6363 processed 14433 requests in 1.000 seconds for a rate of 14433 GETS/sec.
+with an average latency of 2.997 msecs
+```
 
-big bin> hose -t 2 -c 20 -p v1/Article/11 localhost:6363
-localhost:6363 did not respond to 1 requests.
-localhost:6363 processed 17569 requests in 1.000 seconds for a rate of 17569 GETS/sec.
-with an average latency of 2.963 msecs
+Keep the connection alive it were the same browsers performing multiple fetchs.
```
Comment (ashmaroli) [inline examples/sample/README.md]: need an `if` somewhere in there.. :smiley:

File: examples/sample/README.md [hunk 11]
```diff
@@ -173,35 +187,74 @@ connections at a time to the Runner.
...
 
 ```
+> hose -t 2 -c 20 -p e1/Article/11 127.0.0.1:6363   
+127.0.0.1:6363 processed 137603 requests in 1.000 seconds for a rate of 137603 GETS/sec.
+with an average latency of 0.247 msecs
+
+
+```
+
+Keep the connection alive it were the same browsers performing multiple fetchs.
```
Comment (ashmaroli) [inline examples/sample/README.md]: ..and here..

**Comment by ohler55** [comment]
I was very happy with the benchmarks. Better than I originally expected.

**Comment by ashmaroli** [comment]
:smiley:  :+1: Congratulations!

**Pull request merged**